app/components/post_body/post_body.js

```diff
@@ -66,6 +67,18 @@ class PostBody extends PureComponent {
         toggleSelected: emptyFunction
     };
 
+    constructor(props) {
+        super(props);
+
+        this.link = extractFirstLink(props.message);
```

Comment:
What are the benefits of not setting this in the state?

Reply:
Avoid unnecesary re-render

Reply:
Won't it only change when the message has already changed though? It won't cause a re-render on its own. Plus we're using it in the render function, so it should be in the state to keep it pure